lib/Service/ContactsIntegration.php

```diff
@@ -26,40 +26,67 @@
 
 use OCP\Contacts\IManager;
 use OCP\IConfig;
+use OCP\IGroupManager;
 
 class ContactsIntegration {
 
 	/** @var IManager */
 	private $contactsManager;
 
+	/** @var IGroupManager */
+	private $groupManager;
+
 	/** @var IConfig */
 	private $config;
 
-	public function __construct(IManager $contactsManager, IConfig $config) {
+	public function __construct(IManager $contactsManager, 
+								IGroupManager $groupManager,
+								IConfig $config) {
 		$this->contactsManager = $contactsManager;
+		$this->groupManager = $groupManager;
 		$this->config = $config;
 	}
 
 	/**
 	 * Extracts all matching contacts with email address and name
 	 *
+	 * @param string $userId
 	 * @param string $term
 	 * @return array
 	 */
-	public function getMatchingRecipient(string $term): array {
+	public function getMatchingRecipient(string $userId, string $term): array {
 		if (!$this->contactsManager->isEnabled()) {
 			return [];
 		}
 
 		// If 'Allow username autocompletion in share dialog' is disabled in the admin sharing settings, then we must not
 		// auto-complete system users
 		$allowSystemUsers = $this->config->getAppValue('core', 'shareapi_allow_share_dialog_user_enumeration', 'no') === 'yes';
+		$allowSystemUsersInGroupOnly = $this->config->getAppValue('core', 'shareapi_restrict_user_enumeration_to_group', 'no') === 'yes';
+		if ($allowSystemUsersInGroupOnly) {
+			$userGroups = $this->groupManager->getUserIdGroupIds($userId);
```

Comment:
This method is private.

Inject the `IUserManager` into the constructor and get the user object:

```suggestion
		$user = $this->userManager->get($userId);
		if($user === null) {
			return [];
		}
			$userGroups = $this->groupManager->getUserGroupIds($user);
```